client/site-profiler/utils/validate-domain.ts

```diff
@@ -0,0 +1,27 @@
+export default function validateDomain( value: string ) {
```

Comment:
There is a similar function here: https://github.com/Automattic/wp-calypso/blob/update/site-profiler-36/client/site-profiler/utils/normalize-whois-url.ts#L4 used for URLs found in the WHOIS.

Maybe we can create one the we can use with both? The WHOIS normalizer can create this new utility here after the first `normalizeWhoisField`. What do you think?

Reply:
I'm always for that to share and reuse the code. Those two functions, indeed, are similar.
However, they are only sharing the logic of initializing the URL object and catching errors if it is invalid.

I don't see a beneficial way of refactoring those two functions (if start doing it, we should find better naming as well).

If you don't mind, I would leave it for the next iteration of improvements since this PR has a lot of changes, and I tested it in detail. I don't want to make a regression.

Thanks for reviewing and testing this PR, I know it wasn't easy!!!